### PR TITLE
Support for out of process configuration clients

### PR DESCRIPTION
**Pull request by JohnMcPMS**
## Change
The goal of this rather large change is to support out-of-process clients to configuration.  This enables supported scenarios to not require shipping the entire PowerShell infrastructure, and instead leverage the one that is contained in our package.

The interface changes are breaking, but minimal.  The goal is to remove the necessity for a processor to need the configuration module itself, while creating a single entry point that can be used for OOP cases (and many in-proc cases as well).

## Validation
New tests are added for configuration via `winget.exe`, and the existing configuration unit tests were updated to support running both in and out of process (where appropriate).
 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-cli/pull/3363&drop=dogfoodAlpha

### CODE REVIEW

File: src/Microsoft.Management.Configuration.OutOfProc/Factory.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include <atomic>
+
+namespace Microsoft::Management::Configuration::OutOfProc
+{
+    struct Factory : winrt::implements<Factory, winrt::Windows::Foundation::IActivationFactory, IClassFactory>
+    {
+        Factory();
+        ~Factory();
+
+        // Determines if the reference count is 0.
```
Comment (msftrubengu) [inline src/Microsoft.Management.Configuration.OutOfProc/Factory.h:16]: is not 0?

File: src/AppInstallerSharedLib/Security.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        THROW_HR(E_UNEXPECTED);
+    }
+
+    bool IsCOMCallerSameUserAndIntegrityLevel()
+    {
+        auto serverUser = wil::get_token_information<TOKEN_USER>();
+        IntegrityLevel serverIntegrityLevel = GetEffectiveIntegrityLevel();
+
+        ImpersonateCOMorRPCCaller impersonation;
```
Comment (msftrubengu) [inline src/AppInstallerSharedLib/Security.cpp:90]: I got confused here... can you add a comment or name this var startImpersonation?

**Review by msftrubengu — DISMISSED** [review]

File: src/Microsoft.Management.Configuration.Processor/Microsoft.Management.Configuration.Processor.csproj [hunk 3]
```diff
@@ -17,6 +17,7 @@
     <!-- Disable warning CS1591 for cswinrt auto-generated files and CS8785 for SourceGenerator compilation errors -->
     <NoWarn>1591,8785</NoWarn>
     <Configurations>Debug;Release;ReleaseStatic</Configurations>
+    <CsWinRTEnableLogging>true</CsWinRTEnableLogging>
```
[the remark this answers is not in the document]
Comment (JohnMcPMS) [reply src/Microsoft.Management.Configuration.Processor/Microsoft.Management.Configuration.Processor.csproj:20]: Not really, the file is only a few KB.

File: src/Microsoft.Management.Configuration.OutOfProc/Factory.h (hunk 1 above)
Comment (JohnMcPMS) [reply src/Microsoft.Management.Configuration.OutOfProc/Factory.h:16]: It is a boolean, you get to know both whether it is 0 AND whether it is not 0.

**Review by msftrubengu — APPROVED** [review]

**Pull request merged**